### ISSUE TITLE
Update for 345M

### ISSUE THREAD

**Issue by david-sg** [minimaxir/gpt-2-simple#23]
Openai released the 354mb model 
https://github.com/openai/gpt-2/blob/master/DEVELOPERS.md

Could you update to that version?

**Comment by minimaxir** [comment]
Oh boy.

In theory this requires zero code changes (pass `model_name="345M"` to `download_gpt2()` and `finetune`) but I'll double check. I'll need to add `model_name` to the CLI as well.

**Comment by veqtor** [comment]
Please add gradient checkpointing as in nshepperds fork
https://github.com/openai/gradient-checkpointing
https://github.com/nshepperd/gpt-2/blob/finetuning/train.py

Ideally, gradient checkpointing needs to be applied in accumulate.py too, so that gradient accumulation works in conjunction with it.

**Comment by minimaxir** [comment]
Yes, adding that. Changes will be made in the `345M` branch before pulling into `master`.

**Comment by minimaxir** [comment]
Completed in #24
